Rakefile

```diff
@@ -13,42 +13,55 @@ auto_parallel_tests = lambda do
   use_auto_parallel_tests
 end
 
+nproc = lambda do
+  # Limit to 6 processes, as higher number results in more time
+  `(nproc 2> /dev/null) || sysctl -n hw.logicalcpu`.to_i.clamp(1, 6).to_s
```

Comment:
You sure you want to make weird output yield a zero via to_i vs Integer()? It'll kinda work but a watchful eye would be required to see that one is subjected to a default/error default behavior. Were it me, I'd want it to crash, and then fix it.

Reply:
I'm fairly sure the command only works on Mac and Linux.  I don't think we should crash on other operating systems.  It's safe to default to a single CPU in that case, even if it is slower.  Maybe we can accept the value via an environment variable (e.g. `NCPU`) to allow parallel testing on other operating systems (or to allow a non-default concurrency limit on Mac or Linux).  What are your thoughts on that?  That should be done in a separate pull request.

Reply:
I submitted pull request #2490 to switch the implementation to use `Etc.nprocessors` (from the `etc` standard library).